app/services/solidus_square/payment_sync_service.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module SolidusSquare
+  class PaymentSyncService < Base
+    attr_reader :params
+
+    def initialize(params)
+      @params = params
+
+      super()
+    end
+
+    def call
+      create_square_payment! if hosted_checkout?
```

Comment:
If the order is not an hosted_checkout we don't have to update the payment and neither to advance the order.

Reply:
https://github.com/nebulab/solidus_square/pull/43/commits/abea86ce4b352a0e46b124beeecf97797b17300c